Mathlib/Topology/EMetricSpace/Defs.lean

```diff
@@ -538,8 +537,16 @@ end Compact
 
 end EMetric
 
---namespace
-/-- We now define `EMetricSpace`, extending `PseudoEMetricSpace`. -/
+/-- Extended metric spaces, with an extended distance `edist` possibly taking the value `∞`.
+
+Each `EMetricSpace` induces a canonical `UniformSpace` and hence a canonical `TopologicalSpace`.
+This is enforced in the type class definition, by extending the `UniformSpace` structure. When
+instantiating a `PseudoEMetricSpace` structure, the uniformity fields are not necessary, they
```

Comment:
This doesn't quite make sense as a docstring on this decl, since it doesn't immediately extend UniformSpace. What about adding a clause to this sentence that it does these things by virtue of extending `PseudoEMetricSpace`.

Reply:
Then we'll end up in a situation where the docstring for EMetricSpace is just a find/replace on the docstring for PseudoEMetricSpace, which is definitely not what we want. Have you considered writing a docstring here instead of copying it from something else?

Reply:
Well I think it's easier to maintain two docstrings that look the same rather than two docstrings that are different for no deep reason. That's called "uniformisation".

In fact, I fail to understand why the docstring for `PseudoMetricSpace` is any different to the one for `PseudoEMetricSpace`.

Reply:
Docstrings on different declarations should be different: the point is to explain what the declarations are doing. If the best description you have for them both is identical, then they shouldn't be different declarations at all. Uniformisation in terms of format is good, but it's actively harmful in terms of content when it comes to documentation.